src/content/docs/reference/Uniforms/Matrices/gbufferModelView.mdx

```diff
@@ -9,4 +9,4 @@ sidebar:
     variant: caution
 ---
 
-### `uniform ;`
\ No newline at end of file
+### `uniform mat4 gbufferModelView;`
```

Comment:
I think these need a little more detail, e.g. converts what space to what space.

Reply:
I wasn't quite sure how they worked, so I left them unfinished, but I think I've figured it out.

`modelViewMatrix` and `projectionMatrix` are the matrices of the **current** program, so `modelViewMatrix == gbuffersModelView` in gbuffers programs, and `modelViewMatrix == shadowModelView` in shadow programs. They are also undefined in all other programs, while `gbuffers-` and `shadow-` matrices seem to always be defined.

Should I add that to the documentation?

Reply:
Close, but not technically correct.
- `modelViewMatrix` turns model space into view space, whereas `gbufferModelView` turns player space into view space. For example, in gbuffers_entities, `modelViewMatrix` (and `gl_ModelViewMatrix`) is just an identity matrix since model space for entities is view space. It is good practice to not assume what model space is and always use the model view matrix to turn it into view space first.
- `projectionMatrix` (and `gl_ProjectionMatrix`) are usually the same as `gbufferProjection`, except for gbuffers_hand (and gbuffers_hand_water) where there is a multiplier on the z axis (`MC_HAND_DEPTH`) so that hand geometry doesn't clip into terrain.

The following compatibility profile matrices are available:
- `gl_ModelViewMatrix` == `modelViewMatrix`
- `gl_ProjectionMatrix` == `projectionMatrix`
- `gl_ModelViewProjectionMatrix` (combines the two above, not really necessary to document)
- `gl_NormalMatrix` (already documented)
- `gl_TextureMatrix[0]` == `textureMatrix`
- `gl_TextureMatrix[1]`, the lightmap matrix (in Iris this should always be applied with `(gl_TextureMatrix[1] * gl_MultiTexCoord1).xy`, unlike in Optifine where's it's a const it may not be that way in Iris according to IMS)

Reply:
I added all the compatibility profile matrices except for `gl_ModelViewProjectionMatrix` and `gl_TextureMatrix[1]` (since they aren't available as uniforms), as well as descriptions to the unfinished ones based on what you said and [this](https://shaderlabs.org/mw/images/5/5a/Space_conversion_cheat_sheet.png).

Hope I got it right this time :)

Reply:
Looks good, just see that one small addition to gbufferModelView